lib/features/caching/config/hive_cache_client.dart

```diff
@@ -77,6 +79,21 @@ abstract class HiveCacheClient<T> {
     });
   }
 
+  Future<List<T>> getListByCollectionId(String collectionId) {
+    return Future.sync(() async {
+      final boxItem = encryption ? await openBoxEncryption() : await openBox();
+      return boxItem.toMap()
+        .where((key, value) {
+          final tupleKey = TupleKey.fromString(key as String);
+          return tupleKey.parts.length >= 2 && tupleKey.parts[1] == collectionId;
+        })
+        .values
+        .toList();
+    }).catchError((error) {
+      throw error;
+    });
+  }
+
```

Comment:
So interesting, please write the ADR about the `collection`. And the reason why we need `TupleKey`.